### ISSUE TITLE
False positive: async jobs

### ISSUE THREAD

**Issue by bluikko** [ansible-community/ansible-lint#1588]
##### Summary

A `command` module task that is run as an async job is incorrectly treated as a normal sync task.

For async tasks the options like `changed_when` (and `failed_when` and so on) are not given to the async `command` task itself, they are given to the `async_status` module task that is run after the async task.

Ansible-lint does not understand this and complains for rule `no-changed-when` for the `command` task.

Example:
```yaml
---
- name: Asynchronous long task
  command: alongtask.sh
  async: 1000
  poll: 0
  register: job_sleeper

- name: Wait for asynchronous job to end
  async_status:
    jid: '{{ job_sleeper.ansible_job_id }}'
  register: job_result
  until: job_result.finished
  retries: 100
  delay: 10
  changed_when: [....]
```

Note how the `changed_when` is given in the `async_status` task and not in the `command` task.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
2.9.21

ansible-lint --version
5.0.8

```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

EL7.9 all updated

##### STEPS TO REPRODUCE

```yaml
---
- name: Asynchronous yum task
  command: alongtask.sh
  async: 1000
  poll: 0
  register: job_sleeper

- name: Wait for asynchronous job to end
  async_status:
    jid: '{{ job_sleeper.ansible_job_id }}'
  register: job_result
  until: job_result.finished
  retries: 100
  delay: 10
  changed_when: [....]
```

```console (paste below)

```

##### Desired Behaviour

Ansible-lint should not detect `no-changed-when` for `command` module task run as async job since the `changed_when` cannot be given to the `command` module task itself.

It should detect that there is a `changed_when` in the following `async_status` task.

##### Actual Behaviour

Ansible-lint detects false positive `no-changed-when` for `command` module task run as async job even though `changed_when` cannot be correctly given for an async task - the `changed_when` is given for the subsequent `async_status` module task.

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
Please provide links to documentation or tickets that support what you reported. The linter is not able to look at different tasks so the best thing we could do is to disable the rule when async is used.

**Comment by bluikko** [comment]
> Please provide links to documentation or tickets that support what you reported. The linter is not able to look at different tasks so the best thing we could do is to disable the rule when async is used.

Support what? That `changed_when` is not given to `command` module?

I found this one: https://stackoverflow.com/questions/36999933/check-stdout-of-async-ansible-task

I do not know where it is documented, if anywhere. I have just implemented this myself and the `changed_when` on the `async_status` task works exactly as it should: the `async_status` task gets the "changed" status correctly: equivalent to `job_result` variable in the OP example.

**Comment by bluikko** [comment]
Here is an example successful run which shows that the original async `command` module task returns only `ansible_job_id` and other async job related variables, no stdout or any other common variables.

It is analogous to the OP example, first an async `command` task and then `async_status` to get the status.

I am surprised I need to defend this, it is not common knowledge?

```console
2021-05-27 12:43:05,824 p=24509 u=username n=ansible | TASK [role : async yum] *********************************************************************
**********************************************************
2021-05-27 12:43:05,828 p=24509 u=username n=ansible | changed: [host.example.com] => changed=true
  ansible_job_id: '766567162536.7150'
  finished: 0
  results_file: /root/.ansible_async/766567162536.7150
  started: 1
2021-05-27 12:43:26,994 p=24509 u=username n=ansible | TASK [role : async job wait] ******************************************************************************************
**********************************************************
2021-05-27 12:43:27,014 p=24509 u=username n=ansible | changed: [host.example.com] => changed=true
  ansible_job_id: '766567162536.7150'
  attempts: 3
  cmd:
  - /usr/bin/yum
  - -y
  - update
  delta: '0:00:14.860299'
  end: '2021-05-27 12:43:20.747460'
  finished: 1
  rc: 0
  start: '2021-05-27 12:43:05.887161'
  stderr: ''
  stderr_lines: <omitted>
  stdout: |-
    Loaded plugins: fastestmirror, post-transaction-actions
    Determining fastest mirrors
    Resolving Dependencies
    --> Running transaction check
[snip]
```

**Comment by bluikko** [comment]
> Thanks for all the details, the bug is at https://github.com/ansible-community/ansible-lint/blob/master/src/ansiblelint/rules/CommandHasChangesCheckRule.py#L50-L56 and can easily be fixed.

I will take a look at later time - I did check the file and didn't see an obvious example how to check that `poll` equals to 0.

Otherwise it looks straightforward enough and I will look for an example to use.

**Comment by bluikko** [comment]
It is my first attempt at making rules so I hope the PR is correct.

**Comment by Red-GV** [comment]
Hello, I think there's been a regression of this issue. I'm seeing this same problem in the latest version of `ansible-lint`. Was this dropped?
